MediaBrowser.Providers/MediaInfo/AudioFileProber.cs

```diff
@@ -157,60 +158,24 @@ public class AudioFileProber
-                var tagTypes = file.TagTypesOnDisk;
+            var test = ATL.Settings.DisplayValueSeparator;
+            Track track = new Track(audio.Path);
 
-                if (tagTypes.HasFlag(TagTypes.Id3v2))
-                {
-                    tags = file.GetTag(TagTypes.Id3v2);
-                }
-                else if (tagTypes.HasFlag(TagTypes.Ape))
-                {
-                    tags = file.GetTag(TagTypes.Ape);
-                }
-                else if (tagTypes.HasFlag(TagTypes.FlacMetadata))
-                {
-                    tags = file.GetTag(TagTypes.FlacMetadata);
-                }
-                else if (tagTypes.HasFlag(TagTypes.Apple))
-                {
-                    tags = file.GetTag(TagTypes.Apple);
-                }
-                else if (tagTypes.HasFlag(TagTypes.Xiph))
-                {
-                    tags = file.GetTag(TagTypes.Xiph);
-                }
-                else if (tagTypes.HasFlag(TagTypes.AudibleMetadata))
-                {
-                    tags = file.GetTag(TagTypes.AudibleMetadata);
-                }
-                else if (tagTypes.HasFlag(TagTypes.Id3v1))
-                {
-                    tags = file.GetTag(TagTypes.Id3v1);
-                }
-            }
-            catch (Exception e)
+            // ATL will fall back to filename as title when it does not understand the metadata
+            if (track.MetadataFormats.All(mf => mf.Equals(ATL.Factory.UNKNOWN_FORMAT)))
             {
-                _logger.LogWarning(e, "TagLib-Sharp does not support this audio");
+                track.Title = mediaInfo.Name;
             }
 
-            tags ??= new TagLib.Id3v2.Tag();
-            tags.AlbumArtists ??= mediaInfo.AlbumArtists;
-            tags.Album ??= mediaInfo.Album;
-            tags.Title ??= mediaInfo.Name;
-            tags.Year = tags.Year == 0U ? Convert.ToUInt32(mediaInfo.ProductionYear, CultureInfo.InvariantCulture) : tags.Year;
-            tags.Performers ??= mediaInfo.Artists;
-            tags.Genres ??= mediaInfo.Genres;
-            tags.Track = tags.Track == 0U ? Convert.ToUInt32(mediaInfo.IndexNumber, CultureInfo.InvariantCulture) : tags.Track;
-            tags.Disc = tags.Disc == 0U ? Convert.ToUInt32(mediaInfo.ParentIndexNumber, CultureInfo.InvariantCulture) : tags.Disc;
+            track.Album = string.IsNullOrEmpty(track.Album) ? mediaInfo.Album : track.Album;
+            track.Year ??= mediaInfo.ProductionYear;
+            track.TrackNumber ??= mediaInfo.IndexNumber;
+            track.DiscNumber ??= mediaInfo.ParentIndexNumber;
 
             if (audio.SupportsPeople && !audio.LockedFields.Contains(MetadataField.Cast))
             {
                 var people = new List<PersonInfo>();
-                var albumArtists = tags.AlbumArtists;
+                var albumArtists = string.IsNullOrEmpty(track.AlbumArtist) ? mediaInfo.AlbumArtists : track.AlbumArtist.Split(InternalValueSeparator);
```

Comment:
Should we combine the two potential lists and then make it distinct?
I haven't looked at how audio files are typically tagged

Reply:
> Should we combine the two potential lists and then make it distinct?

I prefer not to do this because the ffprobe's result will do its own magic which we are going to avoid, and if you want to use ffprobe's result you should not use the atl's result at all because atl and ffprobe may split artists differently which will still cause duplicated entries. The real solution to this is to add a user-configurable setting to let the user decide how the prober should split the tag to get multiple values, but that is out of the scope of this PR.